src/platforms/common/performance/instrumentation/custom-instrumentation.mdx

```diff
@@ -33,84 +33,106 @@ redirect_from:
+```javascript
+function doWork() {
+  const result = Sentry.startSpan({ name: "Parent Span" }, () => {
+    return expensiveParentFunction();
 
-To add custom performance data to your application, you need to create and use spans. Spans are a way to measure the time it takes for a specific action to occur. For example, you can create a span to measure the time it takes for a function to execute.
+    const res = Sentry.startSpan({ name: "Child Span" }, () => {
+      return expensiveChildFunction();
+    });
+  });
+}
+```
 
-To start measuring timing data, you first need to import the SDK.
+## Creating Spans with Decorators
 
-<PlatformContent includePath="enriching-events/import" />
+It’s common to have a single span track the execution of an entire function. In that scenario, there is a decorator you can use to reduce code:
 
-<PlatformContent includePath="performance/span-api-version" />
+```javascript
+???
 
-## Create Active Span
+// TBD
+```
 
-By default created spans are considered active, which means they are put on the Sentry scope. This allows child spans and Sentry errors to be associated with that span. This is the recommended way to create spans.
+Use of the decorator is equivalent to creating the span inside do_work() and ending it when do_work() is finished.
 
-<PlatformContent includePath="performance/add-active-span" />
+The function name will be used as the spans name.
 
-## Get Active Span
+## Creating Spans That Need Manual Finishing
 
-You can also get the current active span, which is useful for when you need to add new child spans.
+Sometimes you do not want to use context managers, and do want to finish spans on your own. You can do this:
 
-<PlatformContent includePath="performance/get-span" />
+```javascript
+const parentSpan = Sentry.startSpan({ name: "Parent Span" });
 
-## Start Independent Spans
+parentWork();
 
-If you want to add a span that is not active, you can create a independent spans. This is useful for when you have work that is grouped together under a single parent span, but is independent from the current active span. In most cases you'll want to create and use active spans.
+const childSpan = Sentry.startSpan({ name: "Child Span" });
 
-<PlatformContent includePath="performance/add-independent-span" />
+childWork();
 
-## Start Transaction
+childSpan.finish();
+parentSpan.finish();
+```
 
-The root span (the span that is the parent of all other spans) is known as a transaction in Sentry. This can be accessed and created separately if you need more control over the timing data or if you use a version of the SDK that does not support the top level span APIs.
+If you want to create independend spans that are not in a parent/child relation ship a as above you can create them like this:
 
-<PlatformContent includePath="performance/enable-manual-instrumentation" />
+```javascript
+const span1 = Sentry.startSpan({ name: "first-step", independend: true });
```

Comment:
I prefer not to make this an argument because the behavior changes quite a bit between the two modes.

Active span mode **requires** the use of a callback (to wrap) and returns the result of the callback - not the span.

Independent span mode returns just the span.

Certain languages cannot overload arguments/return values like this.